test/controllers/trips_controller_test.rb

```diff
@@ -1,23 +1,266 @@
+      # Arrange
+      @passenger = Passenger.create(name: "Judy", phone_num: "360-555-0987")
+      
+      # Act 
+      expect {
+        post passenger_trips_path(@passenger.id)
+      }.must_differ 'Trip.count', 1
+
+      # Check the rating for new trip is nil and driver available becomes false
+      trip = @passenger.trips.last
+      expect(trip.rating).must_be_nil
+      expect(trip.driver.available).must_equal false
+
+      # Check that the controller redirected the user
+      must_respond_with :redirect
+      must_redirect_to passenger_path(@passenger.id)
+    end
+
+
+    it "does not create a trip if no driver is available" do
+      # Arrange
+      @passenger = Passenger.create(name: "Judy", phone_num: "360-555-0987")
+      available_driver.update(available: false)
+      Driver.all.update(available: false)
+
+      # Act
+      expect {
+        post passenger_trips_path(@passenger.id)  
+      }.wont_change 'Trip.count'
+
+      # Assert
+      must_redirect_to passenger_path(@passenger.id)
+    end
+
+    it "does not create a trip for non-existing passenger" do
+      passenger_id = -1
+
+      expect {
+        post passenger_trips_path(passenger_id)
+      }.wont_change 'Trip.count'
+
+      must_respond_with :not_found
+    end
   end
 
   describe "edit" do
-    # Your tests go here
+    before do
+      Driver.create(name: "TEST123", vin: "WBWSS52P9NEYLVDE9", available: true)
+      @passenger = Passenger.create(name: "Judy", phone_num: "360-555-0987")
+      @trip = Trip.create(driver_id: Driver.first.id, passenger_id: Passenger.first.id, date: "2016-04-05", rating: 3, cost: 1293)
+    end
+    it "responds with success when getting the edit page for an existing, valid trip" do
+      # Arrange & Act
+      get edit_trip_path(@trip.id)
+
+      # Assert
+      must_respond_with :success
+    end
+
+    it "responds with redirect when getting the edit page for a non-existing trip" do
+      # Arrange
+      # Act
+      get edit_trip_path(-1)
+
+      # Assert
+      must_respond_with :not_found
+    end
   end
 
   describe "update" do
-    # Your tests go here
+    before do
+      Driver.create(name: "TEST123", vin: "WBWSS52P9NEYLVDE9", available: true)
+      @passenger = Passenger.create(name: "Judy", phone_num: "360-555-0987")
+      @trip = Trip.create(driver_id: Driver.first.id, passenger_id: Passenger.first.id, date: "2016-04-05", rating: 3, cost: 1293)
+    end
+    let (:valid_trip) { 
+      { 
+          trip: {
+          date: "2016-04-05", 
+          rating: 5, 
+          cost: 12.0
+        }
+      }    
+    }
+
+    let (:invalid_trip) {
+      {
+          trip: {
+              date: "2016-04-05",
+              rating: 50,
+              cost: 12.0
+          }
+      }
+    }
+    it "can update an existing trip with valid information accurately, and redirect" do
```

Comment:
I would also include a test that the driver's average rating changes appropriately.